pkg/restapi/issuer/operation/operations.go

```diff
@@ -1176,22 +1193,68 @@ func (o *Operation) storeUserInvitationMapping(mapping *UserInvitationMapping) e
 }
 
 // TODO fulfillment shouldn't be saved, need re-design [Issue#560]
-func (o *Operation) saveCredentialFulfillment(thID string, fulfillment *verifiable.Presentation) error {
-	fbytes, err := fulfillment.MarshalJSON()
+func (o *Operation) saveCredentialAttachmentData(thID string, attachments credentialAttachments) error {
+	manifestBytes, err := json.Marshal(attachments.ManifestID)
 	if err != nil {
-		return fmt.Errorf("failed to marshal credential fulfillment: %w", err)
+		return fmt.Errorf("failed to marshal credential manifest: %w", err)
 	}
 
-	err = o.txnStore.Put(thID, fbytes)
+	fulfilmentBytes, err := json.Marshal(attachments.FulFillment)
+	if err != nil {
+		return fmt.Errorf("failed to marshal credential fulfilment: %w", err)
+	}
+
+	err = o.txnStore.Batch([]storage.Operation{
+		{
+			Key:   manifestDataPrefix + thID,
+			Value: manifestBytes,
+		},
+		{
+			Key:   fulfillmentDataPrefix + thID,
+			Value: fulfilmentBytes,
+		},
```

Comment:
any reason for saving separately as batch? why didn't you save whole `credentialAttachments` by thID as key?

Reply:
The reason to persist with batch feature was when we read fullfilment https://github.com/trustbloc/adapter/blob/3f454b7dda3f1cb5ae33b3212461b4e488cb0cf1/pkg/restapi/issuer/operation/operations.go#L1256

we need fullfilment bytes to create vp, so i had to filter out those , to keep it simpler i chose this approach